drivers/counter/counter_ll_stm32_rtc.c

```diff
@@ -240,11 +241,7 @@ tick_t rtc_stm32_read(const struct device *dev)
 #endif /* CONFIG_COUNTER_RTC_STM32_SUBSECONDS */
 	ARG_UNUSED(dev);
 
-	/* Enable Backup access */
-#if defined(PWR_CR_DBP) || defined(PWR_CR1_DBP) || \
-	defined(PWR_DBPCR_DBP) || defined(PWR_DBPR_DBP)
-	LL_PWR_EnableBkUpAccess();
-#endif /* PWR_CR_DBP || PWR_CR1_DBP || PWR_DBPR_DBP */
+	stm32_backup_domain_enable_access();
```

Comment:
It seems like the refcount won't be balanced at all here. We'll keep incrementing it each time we'll read, and never decrementing it.
Since it is already enabled at init, I think it isn't necessary to keep it here.

Reply:
Indeed, thanks!

Reply:
Addressed